website/docs/configuration/overview.html.md

```diff
@@ -0,0 +1,182 @@
+---
+layout: "language"
+page_title: "Configuration Language"
+sidebar_current: "docs-config-index"
+description: |-
+  Terraform uses text files to describe infrastructure and to set variables.
+  These text files are called Terraform _configurations_ and are
+  written in the Terraform language.
+---
+
+# Configuration Language
+
+-> **Note:** This page is about Terraform 0.12 and later. For Terraform 0.11 and
+earlier, see
+[0.11 Configuration Language](../configuration-0-11/index.html).
+
+> **Hands-on:** Try the [Terraform: Get Started](https://learn.hashicorp.com/collections/terraform/aws-get-started?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS) collection on HashiCorp Learn.
+
+Terraform uses its own configuration language, designed to allow concise
+descriptions of infrastructure. The Terraform language is declarative,
+describing an intended goal rather than the steps to reach that goal.
+
+## Resources and Modules
+
+The main purpose of the Terraform language is declaring [resources](./resources.html).
+All other language features exist only to make the definition of resources
+more flexible and convenient.
+
+A group of resources can be gathered into a [module](./modules.html),
+which creates a larger unit of configuration. A resource describes a single
+infrastructure object, while a module might describe a set of objects and the
+necessary relationships between them in order to create a higher-level system.
+
+A _Terraform configuration_ consists of a _root module_, where evaluation
+begins, along with a tree of child modules created when one module calls
+another.
+
+## Arguments, Blocks, and Expressions
+
+The syntax of the Terraform language consists of only a few basic elements:
+
+```hcl
+resource "aws_vpc" "main" {
+  cidr_block = var.base_cidr_block
+}
+
+<BLOCK TYPE> "<BLOCK LABEL>" "<BLOCK LABEL>" {
+  # Block body
+  <IDENTIFIER> = <EXPRESSION> # Argument
+}
+```
+
+- _Blocks_ are containers for other content and usually represent the
+  configuration of some kind of object, like a resource. Blocks have a
+  _block type,_ can have zero or more _labels,_ and have a _body_ that contains
+  any number of arguments and nested blocks. Most of Terraform's features are
+  controlled by top-level blocks in a configuration file.
+- _Arguments_ assign a value to a name. They appear within blocks.
+- _Expressions_ represent a value, either literally or by referencing and
+  combining other values. They appear as values for arguments, or within other
+  expressions.
+
+For full details about Terraform's syntax, see:
+
+- [Configuration Syntax](./syntax.html)
+- [Expressions](./expressions.html)
+
+## Code Organization
```

Comment:
This feels like a bit of TMI for the intro page. Do we want to leave this explanation to the "Files and Directories" section and just put an introductory sentence here?

Reply:
I do! But I counted that as partial exploding of existing content, which I was trying to minimize on _this_ PR.